### ISSUE TITLE
aria-required-parent doesn’t respect implicit roles

### ISSUE THREAD

**Issue by drwpow** [dequelabs/axe-core#4141]
### Product

axe-core

### Product Version

4.7.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation

This should be valid:

```html
<menu>
  <li role="menuitem">One</li>
  <li role="menuitem">Two</li>
  <li role="menuitem">Three</li>
</menu>
```

However, it throws the following error:

#### Actual

This error is thrown instead:

```bash
Expected the HTML found at $('li:nth-child(1)') to have no violations:

<li role="menuitem">One</li>

Received:

"Certain ARIA roles must be contained by particular parents (aria-required-parent)"

Fix any of the following:
  Required ARIA parents role not present: menu, menubar, group

You can find more information on this issue here:
https://dequeuniversity.com/rules/axe/4.7/aria-required-parent?application=axeAPI
```

#### How to Reproduce

See above. The issue is fixed by adding:

```diff
- <menu>
+ <menu role="menu">
```

However, this shouldn’t be necessary (in fact, general a11y guidelines is to prefer semantic elements over manual `role` assertions)

#### Additional context

N/A

**Comment by straker** [comment]
Thanks for the issue. So unintuitively the `menu` element's implicit role is not `menu` but instead [is the `list` role](https://www.w3.org/TR/html-aria/#el-menu). This is why you need to add the `role=menu` to it.

**Comment by drwpow** [comment]
![“My disappointment is immeasurable and my day is ruined”](https://github.com/dequelabs/axe-core/assets/1369770/e208e162-d8ed-48af-a2d0-067956a07023)

(thank you for the response)
